### ISSUE TITLE
[webview_flutter] Keyboard language can't change on Android

### ISSUE THREAD

**Issue by owlsometech-kenyang** [flutter/flutter#41089]
My flutter project use webview_flutter v0.3.14+1 and there is a problem when I try to text with Chinese or other keyboard language, the original keyboard always override the correct keyboard language. As you can see below, the chinese keyboard will show and disappear quickly, and you can't even change the language from the keyboard. 

This is what I get in my application:
![itb8d-5o3b1](https://user-images.githubusercontent.com/50938258/65416080-a680e000-de29-11e9-84ad-c3f3087e6913.gif)

Does anyone have any idea or also struggle with the same issue here?

Thanks in advance for your help!

## Logs
```
[✓] Flutter (Channel unknown, v1.7.8+hotfix.4, on Mac OS X 10.14.5 18F132, locale zh-Hant-TW)
    • Flutter version 1.7.8+hotfix.4 at /Users/bigstarpie/development/flutter
    • Framework revision 20e59316b8 (9 weeks ago), 2019-07-18 20:04:33 -0700
    • Engine revision fee001c93f
    • Dart version 2.4.0

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/bigstarpie/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.0, Build version 11A420a
    • CocoaPods version 1.7.5

[✓] iOS tools - develop for iOS devices
    • ios-deploy 1.9.4

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.38.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.4.1

[✓] Connected device (3 available)
    • Android SDK built for x86 • emulator-5554 • android-x86    • Android 9 (API 28) (emulator)
    • macOS                     • macOS         • darwin-x64     • Mac OS X 10.14.5 18F132
    • web                       • web           • web-javascript • Google Chrome 76.0.3809.132 


• No issues found!
```

**Comment by escamoteur** [comment]
It looks you are on an older Flutter version. Could you switch to stable and do a flutter upgrade?

**Comment by owlsometech-kenyang** [comment]
Hello @escamoteur , I have updated the latest stable version, but still have this problem.
```
[✓] Flutter (Channel stable, v1.9.1+hotfix.2, on Mac OS X 10.14.5 18F132, locale zh-Hant-TW)
    • Flutter version 1.9.1+hotfix.2 at /Users/bigstarpie/development/flutter
    • Framework revision 2d2a1ffec9 (2 weeks ago), 2019-09-06 18:39:49 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/bigstarpie/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.0, Build version 11A420a
    • CocoaPods version 1.7.5

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.38.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.4.1

[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

• No issues found!
```

**Comment by escamoteur** [comment]
does this only happen on iOS?
Could you also give it a try on latest master please?

**Comment by owlsometech-kenyang** [comment]
No, this only happen on Android, iOS works perfect.
I switch to v1.10.6-pre.63, but the problem still exists. 
```
[✓] Flutter (Channel master, v1.10.6-pre.63, on Mac OS X 10.14.5 18F132, locale zh-Hant-TW)
    • Flutter version 1.10.6-pre.63 at /Users/bigstarpie/development/flutter
    • Framework revision cc3ca9a916 (23 minutes ago), 2019-09-25 10:57:58 -0400
    • Engine revision 63949eb0fd
    • Dart version 2.6.0 (build 2.6.0-dev.0.0 69b5681546)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/bigstarpie/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.0, Build version 11A420a
    • CocoaPods version 1.7.5

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.38.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.4.1

[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

• No issues found!
```

**Comment by miyakeryo** [comment]
It also happens with Japanese keyboards.

@mklim
Is this the same problem you know below?
https://github.com/flutter/engine/blob/6ac4f9fdf6c0fc5de53a9032a5d81ba0fad59fe6/shell/platform/android/io/flutter/plugin/platform/SingleViewPresentation.java#L285-L289

**Comment by escamoteur** [comment]
Thanks for trying, I'm sorry but we have to wait for the Flutter team now.

**Comment by red282** [comment]
I found a trick to change a language in that situation. When tapping outside the text field, it loses its focus but the keyboard does not disappear🐛. Then tapping inside the text field again, it gets its focus and now the keyboard works well.

**Comment by lutics** [comment]
is any update on this? I have same problem 😭

**Comment by bedryck** [comment]
I have the same problem too. Initially, the keyboard works, but after the application collapses and reopens, the keyboard stops working. Test on Samsung s10e, google pixel 3xl, Redmi note 4

**Comment by bedryck** [comment]
Also, TextField does not update when tapped on the keyboard after the application was collapsed and reopened

**Comment by Hixie** [comment]
I just tried this with the Blogger app and the Greek keyboard and couldn't reproduce. Maybe it has to specifically be a different actual IME implementation?

**Comment by mklim** [comment]
@Hixie our text input logic is different across different webview and Android versions. I don't remember if we have an exact boundary we can tell just from SDK version, but the main bands to test with is as old an SDK version as possible (20), something recent (28), and Q (29). We execute the same way on Q, but the OS itself refactored text input in significant ways that could affect us. It looks like all the `flutter doctor` reports are on SDK 28.

**Comment by timekone** [comment]
Couldn't reproduce this with Cyrillic or Arabic keyboards. webview_flutter v0.3.19, flutter v1.15.3, Gboard, Android 10

**Comment by tvolkert** [comment]
@bparrishMines this is one of four webview issues marked as critical.  How's your bandwidth for being able to look at these?

**Comment by amirh** [comment]
Note that webview keyboard support is [not ready for production use on Android](https://github.com/flutter/plugins/tree/master/packages/webview_flutter#keyboard-support---not-ready-for-production-use), and this issue is just one of the reasons.

With the current allocation webview keyboard is probably not going to be considered stable before Q3 2020.

I think the current state of this issue reflects reality (a far away milestone, even though it's customer critical), though maybe we want to consider removing the `customer: critical` just so it stops surfacing on weekly issue triages?

**Comment by shinsenter** [comment]
Same issue #54648

**Comment by blasten** [comment]
can someone provide a URL that reproduces this issue?

**Comment by dotcink** [comment]
> can someone provide a URL that reproduces this issue?

https://github.com/flutter/plugins/tree/master/packages/webview_flutter/example project can reproduce this issue.

1. Install Gboard app, and add other languages in it, such as Chinese(“简体中文”)
2. Run the above official example project
3. Click the search box in the left drawer of the flutter site, then the keyboard will show, and CANNOT change languages.

**Comment by Hixie** [comment]
We're aware that this is a high priority bug but we are unable to work on it at this time. We continue to look at this bug every week and will update here as soon as we are able to work on it. Sorry for the delay.

**Comment by Hixie** [comment]
(Assigning per conversation with @blasten; @bparrishMines please let me know if I was wrong to assign this to you!)

**Comment by blasten** [comment]
This has been fixed in Flutter 1.22/ WebView 1.0 - use hybrid composition in the Flutter WebView plugin.

https://github.com/flutter/plugins/tree/master/packages/webview_flutter#using-hybrid-composition
